arch/arm/core/Kconfig

```diff
@@ -57,6 +58,7 @@ config ARM_STACK_PROTECTION
 config ARM_SECURE_FIRMWARE
```

Comment:
Could we eliminate ARM_SECURE_FIRMWARE and ARM_NONSECURE_FIRMWARE?
There are equal to TRUSTED_EXECUTION_SECURE and TRUSTED_EXECUTION_NONSECURE if we make the latter option depending on ARCH_HAS_TRUSTED_EXECUTION.

Reply:
> Could we eliminate ARM_SECURE_FIRMWARE and ARM_NONSECURE_FIRMWARE?
> There are equal to TRUSTED_EXECUTION_SECURE and TRUSTED_EXECUTION_NONSECURE if we make the latter option depending on ARCH_HAS_TRUSTED_EXECUTION.

No, we cannot eliminate them. The reason is that TRUSTED_EXECUTION_SECURE is a user option, while ARM_SECURE_FIRMWARE is the internal (hidden) option, used inside arm/core implementation.

The analogy is HW_STACK_PROTECTION (user option) and ARM_STACK_PROTECTION/MPU_STACK_GUARD (internal ARM option)

TRUSTED_EXECUTION_SECURE is supposed _not_ to be an arm-only option.

Reply:
I agree with Ioannis.  It makes sense to have the generic backed by the arch specific one that gets set by the soc/arch configuration.

Reply:
I'm ok with arch specific defines (e.g. for legacy reasons), but do not think they are really needed when the generic ones are in place with same semantics.